### DISCUSSION TITLE
User Authentication for Quarto

### DISCUSSION THREAD

**Discussion by BrighamEaquinto** [quarto-dev/quarto-cli#8393] [Feature Requests]
#### Protect Quarto products behind authentication so that users must log in to see sensitive information

As a huge fan of Quarto, I was recently able to use it to successfully implement a dashboard and website for a client. Everything worked well from its rich features, the ease of development, and the simplicity for the user. The next step I see is user authentication to protect the company's data. I'm more experienced in the data science realm than that of web development so, apologies if this is a simple question. 

My proposal is to introduce a robust user authentication system within Quarto, creating a secure environment where users must log in to access the Quarto website.

In addition, I see a user management system for developers so they can efficiently maintain user accounts. This system would also provide administration tools with the ability to define specific permissions, granting users access to view particular pages and apply filters to manipulate the data visible on each page. This granular control over user access and data visibility would significantly enhance the security and customization of Quarto websites.

In exploring this concept, I came across Streamlit's efforts in a similar direction https://blog.streamlit.io/streamlit-authenticator-part-1-adding-an-authentication-component-to-your-app/. This reference may offer insights and inspiration for the implementation of user authentication and permissions in Quarto.

Thanks!

**Comment by cderv** [comment]
Thanks for reaching out. 

Related existed discussions on this topic : 

* https://github.com/quarto-dev/quarto-cli/discussions/7511
* https://github.com/quarto-dev/quarto-cli/discussions/3816
* https://github.com/quarto-dev/quarto-cli/discussions/3669
* https://github.com/quarto-dev/quarto-cli/discussions/6052

Quarto is an open source publishing system - it produces documents and websites that you can share and host as you like. 

Usually authentication feature are part of the hosting system / platform, and integrated with authentication providers. 

This is especially true for Static Website Generator. For HTML website as example, Quarto will only produce a folder of files that you can serve. No server is run. This is quite different from Streamlit App that you mentioned, as this kind of app is running Python if I am correct. 

Anyhow, you can look at the related discussions for ideas about services that provide authentication for deployments.
